libs/custom-paper-handler/src/driver/driver.test.ts

```diff
@@ -0,0 +1,587 @@
+import { assert } from '@votingworks/basics';
+import { mocks } from '@votingworks/custom-scanner';
+import { Buffer } from 'buffer';
+import { findByIds, WebUSBDevice } from 'usb';
+import { Uint16 } from '@votingworks/message-coder';
+import {
+  GENERIC_ENDPOINT_OUT,
+  REAL_TIME_ENDPOINT_IN,
+  REAL_TIME_ENDPOINT_OUT,
+  getPaperHandlerWebDevice,
+  PaperHandlerDriver,
+  RealTimeRequestIds,
+  ReturnCodes,
+  GENERIC_ENDPOINT_IN,
+} from './driver';
+import { TOKEN, NULL_CODE } from './constants';
+import { PrinterStatus, ScannerStatus } from './sensors';
+import { setUpMockWebUsbDevice } from './test_utils';
+
+/**
+ * message-coder (new) encodes as Buffers, but the prototype driver (legacy) sends messages as Uint8Array.
+ * This means the type of input for calls to webDevice.transferOut will differ between the legacy and new implementations.
+ * We wrap test expectations in formatTransferOutArg now so we can easily change them in the future.
+ */
+function formatTransferOutArg(data: Buffer): Uint8Array | Buffer {
+  return new Uint8Array(data);
+}
+type MockWebUsbDevice = mocks.MockWebUsbDevice;
+
+jest.mock('usb');
+const findByIdsMock = findByIds as jest.MockedFunction<typeof findByIds>;
+const createInstanceMock = WebUSBDevice.createInstance as jest.MockedFunction<
+  typeof WebUSBDevice.createInstance
+>;
+
+let mockWebUsbDevice: MockWebUsbDevice;
+let paperHandlerWebDevice: WebUSBDevice;
+let paperHandlerDriver: PaperHandlerDriver;
+
+export async function setUpMocksAndDriver(): Promise<void> {
+  // setUpMockWebUsbDevice creates a MockWebUsbDevice (A) that represents the paper handler.
+  // getPaperHandlerWebDevice is the driver function that returns the underlying WebUSBDevice (B).
+  // In this case, A and B are the same, but we need to return both because (A) is the mock type and (B)
+  // is the standard WebUSBDevice type. Some tests require both types.
+  const mockDevices = setUpMockWebUsbDevice(findByIdsMock, createInstanceMock);
+  mockWebUsbDevice = mockDevices.mockWebUsbDevice;
+
+  paperHandlerWebDevice = (await getPaperHandlerWebDevice()) as WebUSBDevice;
+  assert(paperHandlerWebDevice);
+
+  paperHandlerDriver = new PaperHandlerDriver(paperHandlerWebDevice);
+  await paperHandlerDriver.connect();
+}
+
+beforeEach(async () => {
+  await setUpMocksAndDriver();
+});
+
+test('initializePrinter sends the correct message', async () => {
+  const transferOutStub = jest.fn();
+  paperHandlerWebDevice.transferOut = transferOutStub;
+
+  await paperHandlerDriver.initializePrinter();
+  expect(paperHandlerWebDevice.transferOut).toHaveBeenCalledTimes(1);
+  expect(paperHandlerWebDevice.transferOut).toHaveBeenCalledWith(
+    GENERIC_ENDPOINT_OUT,
+    formatTransferOutArg(Buffer.from([0x1b, 0x40]))
```

Comment:
I'm only going to update this one, but this change can be made throughout this file.
```suggestion
    Uint8Array.from([0x1b, 0x40])
```

Reply:
[a75568](https://github.com/votingworks/vxsuite/pull/3495/commits/5a75568a30256de39a67867b709847330c7312fd)